src/Compilers/Core/Portable/Diagnostic/DiagnosticDescriptor.cs

```diff
@@ -227,6 +227,8 @@ private static ReportDiagnostic MapSeverityToReport(DiagnosticSeverity severity)
                     return ReportDiagnostic.Warn;
                 case DiagnosticSeverity.Error:
                     return ReportDiagnostic.Error;
+                case DiagnosticSeverity.Suggestion:
```

Comment:
> I think visual presentation (where it shows up and how it shows up) is options for diagnostics which I believe should be just flags. now complexity is linear rather than exponential. (each flag is independent to others rather than we need all combinations)

I don't understand what this means.  How do you do this with flags?  How do you handle incompatible combinations of flags. i.e. what do you do if someone says "I'm a warning and error".

Right now we have severities to represent different, exclusive severity states that the diagnostic can be in.  We have tags to represent things that can augment a diagnostic *regardless* of what severity it is.

'Unnecessary' is an example of a tag/flag.  It adds information to any potential diagnostic, regardless of severity.

Suggestion is not an example of a tag/flag.  It is a distinct state from things like 'Warning' or 'Error'.